doc/source/user_guide/tutorial_parallelization.rst

```diff
@@ -1,11 +1,73 @@
-========================
-How to parallelize loops
-========================
+=========================
+Parallel image processing
+=========================
 
-In image processing, we frequently apply the same algorithm
-on a large batch of images. In this paragraph, we propose to
-use `joblib <https://joblib.readthedocs.io>`_ to parallelize
-loops. Here is an example of such repetitive tasks:
+Sometimes you will need to process batches of images, or images
+that do not fit into memory. Here we discuss how you could use `joblib
+<https://joblib.readthedocs.io>`__ or `dask
+<https://dask.pydata.org>`__ for such tasks.
+
+
+Batch processing with ``joblib``
+--------------------------------
+
+Consider the case where you have several images to be processed, and you want
+to calculate the Otsu threshold in all of them:
+
+.. code-block:: python
+
+    from skimage import io, filters, color
+    import glob
+
+    # Load all images into memory
+    image_files = sorted(glob.glob('*.png'))
+    images = [io.imread(image) for image in image_files]
+
+    def threshold(image):
+        return filters.threshold_otsu(color.rgb2gray(image))
+
+    # Process all images serially and store the result
+    thresholds = [threshold(image) for image in images]
+
+The code above could be improved in two aspects: first, it attempts to load
+*all* image data into memory at once. Second, it processes the images serially,
+not using more than one CPU core.
+
+Instead, we can use ``joblib`` to parallelize the batch processing as follows:
+
+.. code-block:: python
+
+    from joblib import Parallel, delayed
+
+    image_files = sorted(glob.glob('*.png'))
+
+    def load_and_threshold(image_fn):
+        image = io.imread(image_fn)
+        return filters.threshold_otsu(color.rgb2gray(image))
+
+    threshold_tasks = [delayed(load_and_threshold)(fn) for fn in image_files]
+
+    p = Parallel(n_jobs=-1, backend='threading')
+    thresholds = p(threshold_tasks)
+
+The ``delayed`` call prevents ``load_and_threshold`` from executing immediately
+while setting up the parallel processing. Instead, it becomes a task that can
+be scheduled. Setting ``n_jobs`` to ``-1`` tells ``joblib`` to use all cores
+available.
```

Comment:
By experience, I prefer -2, all cores minus one, to do not freeze/overload the computer.